core/client/kt/verify_test.go

```diff
@@ -89,34 +108,53 @@ func TestVerifyGetEntyrResponse(t *testing.T) {
+				VrfProof:  []byte{0x9f, 0x8f, 0xb1, 0x41, 0xbc, 0x10, 0xcb, 0xe2, 0x02, 0xe8, 0x3e, 0x8a, 0xe2, 0xd0, 0xe7, 0xe3, 0xc9, 0xa2, 0x83, 0x94, 0x85, 0xf7, 0xca, 0x8f, 0x33, 0xb6, 0x52, 0x56, 0xb0, 0x76, 0x8e, 0xf9, 0x6e, 0x0c, 0x8a, 0x1c, 0xe6, 0x7d, 0x8b, 0xb6, 0x73, 0xb3, 0xae, 0x51, 0x36, 0x52, 0xab, 0x2b, 0x9d, 0x5a, 0x96, 0xdd, 0xae, 0x2a, 0x74, 0x74, 0x02, 0x6b, 0xdd, 0x16, 0x86, 0x70, 0x94, 0x15, 0x04, 0x7b, 0xcd, 0x07, 0x03, 0xb9, 0x69, 0xe3, 0x72, 0x35, 0xdb, 0xfc, 0xb2, 0xa3, 0x4c, 0x22, 0x6b, 0xaa, 0xce, 0x92, 0x6b, 0xcf, 0x02, 0x11, 0x78, 0x7b, 0x1f, 0x5c, 0x2f, 0xff, 0xb9, 0x34, 0x32, 0xa1, 0xd9, 0xba, 0xec, 0xa5, 0x9d, 0x5e, 0xa6, 0xbb, 0xb6, 0x77, 0x92, 0x4c, 0x0c, 0x2d, 0x76, 0xdf, 0xbe, 0x9e, 0xa0, 0x93, 0xde, 0xf5, 0xa1, 0xc1, 0x4e, 0x9e, 0x19, 0x39, 0x16, 0xfe, 0x60},
 				Committed: nil,
 				LeafProof: &trillian.MapLeafInclusion{
 					Leaf: &trillian.MapLeaf{},
-					Inclusion: append(make([][]byte, 255),
-						[]byte{0xf2, 0x61, 0x90, 0xf8, 0x52, 0x2c, 0xfa, 0x8, 0xfa, 0xc0, 0xb2, 0x54, 0x4c, 0x78, 0x54, 0xad, 0x7, 0x5, 0xe6, 0xec, 0x43, 0xd3, 0xba, 0xed, 0x5f, 0x25, 0x9e, 0x7, 0x1c, 0x63, 0xa6, 0x97}),
+					Inclusion: [][]byte{
+						nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil,
+						nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil,
+						nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil,
+						nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil,
+
+						nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil,
+						nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil,
+						nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil,
+						nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil,
+
+						nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil,
+						nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil,
+						nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil,
+						nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil,
+
+						nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil,
+						nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil,
+						nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil,
+						nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil,
+
+						{92, 215, 13, 113, 97, 138, 214, 158, 13, 29, 227, 67, 236, 34, 215, 4, 76, 188, 79, 247, 149, 223, 227, 147, 86, 214, 90, 126, 192, 212, 113, 64},
```

Comment:
This change is actually expected.  In the new unit tests I exercise the extreme case where a leaf node has a neighbor at the bottom of the tree, which is vanishingly unlikely in practice.   A "normal" inclusion proof will be mostly non-nil values near the top, and nil values near the bottom.

Reply:
I got so stuck in the weeds here that I didn't realise the original test also looked like this!
I prefer the way you construct the array to my sea of nils.  I'll fix that.

Reply:
Done.